ui/src/pages/deployment-details/deployment-details-form/section-location/geo-search/useGeoSearch.ts

```diff
@@ -0,0 +1,57 @@
+import axios from 'axios'
+import { MarkerPosition } from 'design-system/map/types'
+import _ from 'lodash'
+import { useCallback, useEffect, useState } from 'react'
+import { SearchResult, ServerSearchResult } from './types'
+
+const API_URL = '/nominatim'
+const MAX_NUM_RESULTS = 5
+
+const convertServerResult = (result: ServerSearchResult): SearchResult => {
+  const latitude = _.toNumber(result.lat)
+  const longitude = _.toNumber(result.lon)
+
+  return {
+    osmId: result.osm_id,
+    displayName: result.display_name,
+    position: new MarkerPosition(latitude, longitude),
+  }
+}
+
+const getFetchUrl = (searchString: string) =>
+  `${API_URL}?format=json&q=${searchString}`
+
+export const useGeoSearch = (searchString: string) => {
+  const [fetchUrl, setFetchUrl] = useState<string>()
+  const debouncedSetFetchUrl = useCallback(_.debounce(setFetchUrl, 200), [
+    setFetchUrl,
+  ])
+  const [data, setData] = useState<SearchResult[]>()
+  const [isLoading, setIsLoading] = useState(true)
+  const [error, setError] = useState<Error>()
+
+  useEffect(() => {
+    debouncedSetFetchUrl(getFetchUrl(searchString))
+  }, [searchString])
```

Comment:
is the useEffect needed here as the fetchUrl is in state? Will it not get called on each render when searchString (prop) or url (state) changes anyway?

Reply:
Fixed together with cleanup above.